cpp/modmesh/toggle/RadixTree.cpp

```diff
@@ -72,6 +72,143 @@ void CallProfiler::print_profiling_result(const RadixTreeNode<CallerProfile> & n
+            for (const auto & child : current_node->children())
+            {
+                nodes_buffer.push(child.get());
+            }
+        }
+    }
+
+    // Newline after the last element.
+    outstream << '\n';
+    outstream << R"(        ],)" << '\n';
+}
+
+void CallProfilerSerializer::serialize_radix_tree_node(const RadixTreeNode<CallerProfile> & node, bool is_first_node, std::ostream & outstream)
+{
+    // Serialize the RadixTreeNode to the json format.
+
+    // Avoid the trailing comma for the first node.
+    if (!is_first_node)
+    {
+        outstream << R"(,)" << '\n';
+        outstream << R"(            {)" << '\n';
+    }
+    else
+    {
+        outstream << R"({)" << '\n';
+    }
+
+    outstream << R"(                "key": )" << node.key() << R"(,)" << '\n';
+    outstream << R"(                "name": ")" << node.name() << R"(",)" << '\n';
+    CallProfilerSerializer::serialize_caller_profile(node.data(), outstream);
+    outstream << R"(                "address": ")" << reinterpret_cast<std::uintptr_t>(&node) << R"(",)" << '\n';
+    CallProfilerSerializer::serialize_radix_tree_node_children(node.children(), outstream);
+
+    outstream << R"(            })";
+}
+
+void CallProfilerSerializer::serialize_radix_tree_node_children(const CallProfilerSerializer::child_list_type & children, std::ostream & outstream)
+{
+    // Serialize the children list in RadixTreeNode.
+    outstream << R"(                children": [)";
+
+    // If the children list is empty, close the list at the same line.
+    if (children.empty())
+    {
+        outstream << R"(])" << '\n';
+    }
+
+    else
+    {
+        outstream << '\n';
+
+        bool is_first_child = true;
+        for (const auto & child : children)
+        {
+            // Avoid the trailing comma.
+            if (!is_first_child)
+            {
+                outstream << R"(,)" << '\n';
+            }
+            is_first_child = false;
+            outstream << R"(                    ")" << reinterpret_cast<std::uintptr_t>(child.get()) << R"(")";
+        }
+        outstream << '\n';
+        outstream << R"(                ])" << '\n';
+    }
+}
+
+void CallProfilerSerializer::serialize_caller_profile(const CallerProfile & profile, std::ostream & outstream)
+{
+    // Serialize the CallerProfile to the json format.
+    outstream << R"(                "data": {)" << '\n';
+    outstream << R"(                    "start_time": )" << profile.start_time.time_since_epoch().count() << R"(,)" << '\n';
+    outstream << R"(                    "caller_name": ")" << profile.caller_name << R"(",)" << '\n';
+    outstream << R"(                    "total_time": )" << profile.total_time.count() << R"(,)" << '\n';
+    outstream << R"(                    "call_count": )" << profile.call_count << R"(,)" << '\n';
+    outstream << R"(                    "is_running": )" << profile.is_running << '\n';
+    outstream << R"(                },)" << '\n';
+}
+
```

Comment:
1. I add the indent for the output.
2. I remove the lengthy comment.
3. I split more function to improve the readability.